sdk/search/search-documents/review/search-documents.api.md

```diff
@@ -619,6 +619,9 @@ export type HighWaterMarkChangeDetectionPolicy = BaseDataChangeDetectionPolicy &
     highWaterMarkColumnName: string;
 };
 
+// @public (undocumented)
+export type IfEquals<X, Y, A, B> = (<T>() => T extends X ? 1 : 2) extends <T>() => T extends Y ? 1 : 2 ? A : B;
+
```

Comment:
This type is pretty complicated. If we have to have this in the public API, we need to at least document it and I'd suggest the following names for the type and args instead of `IfEquals<X, Y, A, B>`: `IfIdentical<T1, T2, Then, Else>`, but honestly, I think it's a bit much, and we could use simplified expansions inline.

Reply:
It turns out that either JSDoc or API Extractor is very particular about how doc comments are structured. The last line of the docs here was `* */`, and it didn't like that. In any case, I've inlined this type.